src/js/components/homepage/visualizations/treemap/BudgetSubfunctions/BudgetSubfunctionsNavigation.jsx

```diff
@@ -0,0 +1,202 @@
+/**
+ * BudgetSubfunctionsNavigation.jsx
+ * Created by michaelbray on 5/30/17.
+ */
+
+import React from 'react';
+import * as Icons from 'components/sharedComponents/icons/Icons';
+
+import MinimizedBudgetFunctions from './MinimizedBudgetFunctions/MinimizedBudgetFunctions';
+import MinimizedBudgetFunctionsTooltip from
+    './MinimizedBudgetFunctions/MinimizedBudgetFunctionsTooltip';
+
+const propTypes = {
+    alternateColors: React.PropTypes.array,
+    categories: React.PropTypes.object,
+    changeActiveSubfunction: React.PropTypes.func,
+    colors: React.PropTypes.array,
+    descriptions: React.PropTypes.array,
+    selected: React.PropTypes.number,
+    toggleSubfunction: React.PropTypes.func,
+    tooltipStyles: React.PropTypes.object,
+    totalNumber: React.PropTypes.number
+};
+
+const arrows = {
+    all: {
+        direction: 0,
+        label: "All Budget Functions"
+    },
+    previous: {
+        direction: -1,
+        label: "Previous Budget Function"
+    },
+    next: {
+        direction: 1,
+        label: "Next Budget Function"
+    }
+};
+
+export default class BudgetSubfunctionsNavigation extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            label: null,
+            width: 230,
+            height: 44
+        };
+
+        this.showArrowTooltip = this.showArrowTooltip.bind(this);
+        this.createArrowTooltip = this.createArrowTooltip.bind(this);
+    }
+
+    changeFunction(direction) {
+        // 'direction' is -1 for previous, 0 for all (return), and 1 for next
+        if (direction === 0) {
+            this.props.toggleSubfunction();
```

Comment:
Added default value for `selected` in TreeMap's `toggleSubfunction` method to prevent setting state to `undefined`